server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java

```diff
@@ -144,9 +146,16 @@ public CompletableFuture<BatchIterator<Row>> getIterator(Role currentUser,
 
         Settings options = server.options();
         String url = urlSetting.get(options);
+        try {
+            // implicit call to Driver.acceptsURL to make sure there is a driver that accepts the URL
+            DriverManager.getDriver(url);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Why is this necessary here? It would fail later anyway if there is no driver registered for the URI

Reply:
Since we added `substring(5)` it would be needed to filter urls like `foo`.

Reply:
Sorry did not make it clear, for example for `foo` as the url, `Range [5, 3) out of bounds for length 3` is thrown. Also, `foofoo` can trigger
```
Only a super user can connect to localhost unless `fdw.allow_local` is set to true
```
because `uri.getHost()` returns `null`.

Reply:
I'd rather we handle that more excplicit then. E.g. replace the `assert url.startsWith("jdbc:")` with an `if`, and raise a failure if it doesn't start with `jdbc`:

Reply:
Thank you, updated.